build.gradle

```diff
@@ -303,6 +303,20 @@ subprojects {
           } else {
             apply plugin: 'com.github.johnrengelman.shadow'
             artifact shadowJar
+            // work around issue with 'shadow' to resolve runtime dependencies correctly. This requires
+            // any project relying on 'shadow' dependencies.
+            afterEvaluate {
+              pom.withXml { xml ->
+                def dependenciesNode = xml.asNode().appendNode('dependencies')
+                project.configurations.shadow.allDependencies.each {
+                  def dependencyNode = dependenciesNode.appendNode('dependency')
+                  dependencyNode.appendNode('groupId', it.group)
+                  dependencyNode.appendNode('artifactId', it.name)
+                  dependencyNode.appendNode('version', it.version)
+                  dependencyNode.appendNode('scope', 'runtime')
```

Comment:
A fix that involves updating the pom.xml directly doesn't seem right. For example, does that mean that https://docs.gradle.org/current/userguide/publishing_gradle_module_metadata.html is not updated?

Reply:
The problem seems to be that the shadow plugin doesn't respect `archivesBaseName`?

Reply:
Yes @ijuma, I have mostly read the documentation multiple times but do not get the complex behaviour addressed properly. We do already set the archiveClassifier [here](https://github.com/apache/kafka/blob/9e4a4a28218b8803a6f41b65e19a94268035ccf5/build.gradle#L1485) and   archiveBaseName [here](https://github.com/apache/kafka/blob/9e4a4a28218b8803a6f41b65e19a94268035ccf5/build.gradle#L1422), I have also tried setting the `archiveBaseName` inside `shadowJar` definition but it doesn't work with `maven-publication`.